openml/config.py

```diff
@@ -7,47 +7,94 @@
     output_formatter = logging.Formatter(message_format, datefmt='%H:%M:%S')
 
-    console_stream = logging.StreamHandler()
-    console_stream.setFormatter(output_formatter)
-    console_stream.setLevel(verbosity_map[console_output_level])
+    console_handler = logging.StreamHandler()
+    console_handler.setFormatter(output_formatter)
 
-    one_mb = 2**20
+    one_mb = 2 ** 20
     log_path = os.path.join(cache_directory, 'openml_python.log')
-    file_stream = logging.handlers.RotatingFileHandler(log_path, maxBytes=one_mb, backupCount=1)
-    file_stream.setLevel(verbosity_map[file_output_level])
-    file_stream.setFormatter(output_formatter)
+    file_handler = logging.handlers.RotatingFileHandler(
+        log_path, maxBytes=one_mb, backupCount=1, delay=True
+    )
+    file_handler.setFormatter(output_formatter)
+
+
+def _convert_log_levels(log_level: int) -> Tuple[int, int]:
+    """ Converts a log level that's either defined by OpenML/Python to both specifications. """
+    # OpenML verbosity level don't match Python values directly:
+    openml_to_python = {0: logging.WARNING, 1: logging.INFO, 2: logging.DEBUG}
+    python_to_openml = {logging.DEBUG: 2, logging.INFO: 1, logging.WARNING: 0,
+                        logging.CRITICAL: 0, logging.ERROR: 0}
+    # Because the dictionaries share no keys, we use `get` to convert as necessary:
+    openml_level = python_to_openml.get(log_level, log_level)
+    python_level = openml_to_python.get(log_level, log_level)
+    return openml_level, python_level
+
+
+def _set_level_register_and_store(
+        handler: logging.Handler,
+        log_level: int,
+        store_under_key: Optional[str] = None
+):
+    """ Set handler log level, register it if needed, save setting to config file if specified. """
+    oml_level, py_level = _convert_log_levels(log_level)
+    handler.setLevel(py_level)
+
+    if openml_logger.level > py_level or openml_logger.level == logging.NOTSET:
+        openml_logger.setLevel(py_level)
+
+    if handler not in openml_logger.handlers:
+        openml_logger.addHandler(handler)
+
+    if store_under_key is not None:
+        _save_setting_to_configuration(store_under_key, oml_level)
 
-    openml_logger.addHandler(console_stream)
-    openml_logger.addHandler(file_stream)
-    return console_stream, file_stream
+
+def set_console_log_level(
+        console_output_level: Optional[int] = None,
+        save_as_default: bool = False
+):
+    """ Set console output to the desired level and register it with openml logger if needed. """
+    global console_handler
+    store_under_key = 'verbosity' if save_as_default else None
+    _set_level_register_and_store(console_handler, console_output_level, store_under_key)
+
+
+def set_file_log_level(
+        file_output_level: Optional[int] = None,
+        save_as_default: bool = False
+):
+    """ Set file output to the desired level and register it with openml logger if needed. """
```

Comment:
I think "register it with openml logger" is implementation detail. Maybe just saying that "configured logging to a file (if not already) and sets the desired level".